perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/AuthzResolverBlImpl.java

```diff
@@ -2129,11 +2131,19 @@ private static boolean objectAndRoleManageableByEntity(String entityToManage, Pe
 			throw new InternalErrorException("Management rules not exist for the role " + role, e);
 		}
 
-		return createMappingOfValues(complementaryObject, role, rules);
+		//For reading a role it is enough to know role and the primary object.
+		//No need to fetch all related objects
+		Map<String, Integer> mapping = new HashMap<>();
+		Integer role_id = authzResolverImpl.getRoleId(role);
```

Comment:
This is not the correct way how to name variables in Java.

Reply:
Fixed